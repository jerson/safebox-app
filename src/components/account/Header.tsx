import React from 'react';
import {View, Platform, StyleSheet, ViewStyle, TextStyle} from 'react-native';
import Colors from '../../modules/constants/Colors';
import Font from '../../modules/resources/Font';
import Size from '../../modules/dimensions/Size';
import Text from '../ui/Text';
import Icon from 'react-native-vector-icons/Feather';
import {HeaderProps} from 'react-navigation-stack/lib/typescript/types';

const paddingTop = Platform.OS === 'ios' ? Size.getStatusBarHeight() : 0;
const height = Size.getActionBarHeight() + paddingTop;
const styles = StyleSheet.create({
  globalContainer: {
    backgroundColor: Colors.grey1,
    overflow: 'visible',
  } as ViewStyle,
  container: {
    height,
    paddingTop,
    backgroundColor: Colors.grey1,
    flexDirection: 'row',
    alignItems: 'center',
  } as ViewStyle,
  shadow: {
    overflow: 'visible',
  } as ViewStyle,
  title: {
    ...Font({weight: 'Bold'}),
    textAlign: 'center',
    color: Colors.grey6,
    fontSize: 18,
    marginLeft: 10,
    marginBottom: 2,
  } as TextStyle,
  centerContainer: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
  } as ViewStyle,
  leftContainer: {
    width: 60,
  } as ViewStyle,
  rightContainer: {
    width: 60,
  } as ViewStyle,
  icon: {
    fontSize: 20,
  } as TextStyle,
});

export interface AccountHeaderProps {
  title: string;
  icon: string;
  tintColor: string;
}

function Header(props: HeaderProps & AccountHeaderProps) {
  const {options} = props.scene.descriptor;
  const isEmpty = !options.headerLeft && !options.headerRight && !options.title;
  if (isEmpty) {
    return null;
  }

  return (
    <View style={styles.globalContainer}>
      <View style={[styles.container, options.headerStyle, styles.shadow]}>
        <View style={styles.leftContainer}>{options.headerLeft}</View>
        <View style={styles.centerContainer}>
          <Icon
            style={[styles.icon, {color: props.tintColor}]}
            name={props.icon}
          />
          <Text allowFontScaling={false} style={styles.title}>
            {props.title}
          </Text>
        </View>
        <View style={styles.rightContainer}>{options.headerRight}</View>
      </View>
    </View>
  );
}

export default Header;
