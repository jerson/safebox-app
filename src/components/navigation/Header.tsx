import React from 'react';
import { View, Platform, StyleSheet, ViewStyle, TextStyle } from 'react-native';
import { HeaderProps } from 'react-navigation';
import Colors from '../../modules/constants/Colors';
import Font from '../../modules/resources/Font';
import Size from '../../modules/dimensions/Size';
import Text from '../ui/Text';

const paddingTop = Platform.OS === 'ios' ? Size.getStatusBarHeight() : 0;
const height = Size.getActionBarHeight() + paddingTop;
const styles = StyleSheet.create({
  globalContainer: {
    backgroundColor: Colors.grey1,
    overflow: 'visible'
  } as ViewStyle,
  container: {
    height,
    paddingTop,
    backgroundColor: Colors.white,
    flexDirection: 'row',
    alignItems: 'center'
  } as ViewStyle,
  shadow: {
    shadowOpacity: 0.07,
    shadowRadius: 6,
    shadowColor: Colors.black,
    shadowOffset: { height: 2, width: 0 },
    borderBottomLeftRadius: 16,
    borderBottomRightRadius: 16,
    elevation: 4,
    overflow: 'visible'
  } as ViewStyle,
  title: {
    ...Font({ weight: 'Bold' }),
    textAlign: 'center',
    color: Colors.grey6,
    fontSize: 18,
    lineHeight: 20,
    marginTop: 4
  } as TextStyle,
  centerContainer: {
    flex: 1
  } as ViewStyle,
  leftContainer: {
    width: 60
  } as ViewStyle,
  rightContainer: {
    width: 60
  } as ViewStyle
});

function Header(props: HeaderProps) {
  const { options } = props.scene.descriptor;
  const isEmpty = !options.headerLeft && !options.headerRight && !options.title;
  if (isEmpty) {
    return null;
  }

  return (
    <View style={styles.globalContainer}>
      <View style={[styles.container, options.headerStyle, styles.shadow]}>
        <View style={styles.leftContainer}>{options.headerLeft}</View>
        <View style={styles.centerContainer}>
          <Text allowFontScaling={false} style={styles.title}>
            {options.title}
          </Text>
        </View>
        <View style={styles.rightContainer}>{options.headerRight}</View>
      </View>
    </View>
  );
}

export default Header;
