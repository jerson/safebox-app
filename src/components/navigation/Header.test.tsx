import React from 'react';
import * as renderer from 'react-test-renderer';

import Header from './Header';
import { HeaderProps } from 'react-navigation';
import { View } from 'react-native';

const props: HeaderProps = {
  mode: {} as any,
  scene: {
    descriptor: { options: { headerLeft: <View />, title: 'sample' } }
  } as any
} as any;
test('renders without crashing', () => {
  const snapshot = renderer.create(<Header {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<Header {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
