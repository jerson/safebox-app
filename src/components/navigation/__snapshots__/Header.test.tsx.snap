// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafdff",
      "overflow": "visible",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafdff",
          "flexDirection": "row",
          "height": 84,
          "paddingTop": 20,
        },
        undefined,
        Object {
          "overflow": "visible",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "width": 60,
        }
      }
    >
      <View />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontFamily": "Nunito",
              "fontSize": undefined,
              "fontStyle": "normal",
              "fontWeight": "400",
            },
            Object {},
            Object {
              "color": "#38414e",
              "fontFamily": "Nunito",
              "fontSize": 18,
              "fontStyle": "normal",
              "fontWeight": "700",
              "marginBottom": 2,
              "textAlign": "center",
            },
          ]
        }
      >
        sample
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 60,
        }
      }
    />
  </View>
</View>
`;
